runtime/oti/j9nonbuilder.h

```diff
@@ -2967,18 +2967,25 @@ typedef struct J9IndexableObjectContiguous {
 #if defined(J9VM_ENV_DATA64) && defined(OMR_GC_FULL_POINTERS)
 	U_32 padding;
 #endif /* J9VM_ENV_DATA64 && !OMR_GC_COMPRESSED_POINTERS */
+#if defined(J9VM_ENV_DATA64)
```

Comment:
These structures are no longer used (they exist only to keep DDR happy), so if you avoid changing them, then there may not be a need for any DDR work.

Reply:
Sounds good. I used them because I was getting errors on PPC 32 bit machines and the usage of those macros fixed that issue. Which macro should I use to specify if the platform is 32 or 64 bits instead?

Reply:
`J9VM_ENV_DATA64` is the right macro to used.
They must be updated or DDR will break: see my [comment](https://github.com/eclipse/openj9/pull/11507#discussion_r552095630) below.

Reply:
Ah I see what you mean. Only the structures `J9IndexableObject.*Compressed` and `J9IndexableObject.*Full` are used while `J9IndexableObjectContiguous` is not, correct? Can I say the same for `J9IndexableObjectDiscontiguous`?

Reply:
Perhaps the answer is what I thought we had already done - add a flag to indicate the presence of the compressed/full structures and use that instead of the mixed mode flag. Then we could leave the outdated structures as-is.

Reply:
The DDR code must continue to use `J9IndexableObjectContiguous` and `J9IndexableObjectDiscontiguous` to support core files from VMs before the introduction of the `*Compressed` and `*Full` variants. The old types should have sizes that reflect the new reality so the offset to the elements can be correctly computed.

I'm not sure it's worth the effort to avoid using `J9IndexableObjectContiguous` or `J9IndexableObjectDiscontiguous` except for old core files.

The `dataAddr` field should be added to all six types (in 64-bit builds).

Reply:
> The DDR code must continue to use J9IndexableObjectContiguous and J9IndexableObjectDiscontiguous to support core files from VMs before the introduction of the *Compressed and *Full variants.

Curious why we have this restriction? Shouldn't we always use the DDR/`jdmpview` version that came with the VM which generated the core file being investigated?

Reply:
Yes, I suppose there is no benefit to doing as I suggested. The sizes are derived from the core file, so there's no reason not to update the obsolete structures.

Reply:
The intent has always been that DDR is backward-compatible; that is, a new version of `jdmpview` should be able to read a core file produced by an older VM. This greatly simplifies the support story: you don't need to have every version in use by any customer to support them all. If a problem is identified in DDR in the version used by some client, you have the option to use a newer version (perhaps a new, custom version) to examine core files.

Rather than a restriction, this permits greater freedom in your choice of which version of `jdmpview` to use.